core-bundle/src/Search/Document.php

```diff
@@ -146,7 +146,19 @@ static function (Crawler $node) {
         ;
 
         // Filter invalid (null) values
-        $this->jsonLds = array_filter($this->jsonLds);
+        $jsonLds = array_filter($jsonLds);
+
+        // Go through all data entries and check if any array has numeric keys. If yes, it likely contains multiple data inside it
+        // which should be treated as if coming from separate sources, and thus moved to the root of $jsonLds array.
+        foreach ($jsonLds as $jsonLd) {
+            if (array_keys($jsonLd) === range(0, \count($jsonLd) - 1)) {
+                foreach ($jsonLd as $v) {
+                    $this->jsonLds[] = $v;
+                }
+            } else {
+                $this->jsonLds[] = $jsonLd;
+            }
```

Comment:
```suggestion
            $this->jsonLds = array_merge(
                $this->jsonLds,
                ArrayUtil::isAssoc($jsonLd) ? [$jsonLd] : $jsonLd
            );
```

Unit tests work fine for me with `ArrayUtil::isAssoc()`. 
(Suggestion needs additional use statement.)

Reply:
Thanks, improved in 483d0b8.